lib/metasploit/framework/login_scanner/redis.rb

```diff
@@ -64,13 +105,37 @@ def attempt_login(credential)
             )
           end
 
-          disconnect if self.sock
+          disconnect if sock
 
           ::Metasploit::Framework::LoginScanner::Result.new(result_options)
         end
 
         private
 
+        # Authenticates against Redis using the provided credentials arguments.
+        # Takes either a password, or a username and password combination.
+        #
+        # @param [String] username The username to authenticate with, defaults to 'default'
+        # @param [String] password The password to authenticate with.
+        # @return [String] The response from Redis for the AUTH command.
+        def authenticate(username, password)
+          command = redis_proto(['AUTH', username.blank? ? 'default' : username, password])
+          sock.put(command)
+          sock.get_once
+        end
+
+        # Authenticates against Redis using the provided password.
+        # This method is for older Redis instances of backwards compatibility.
+        #
+        # @param [String] password The password to authenticate with.
+        # @return [String] The response from Redis for the AUTH command.
+        def authenticate_backwards(password)
```

Comment:
'backwards' might be confusing to future travellers; Maybe:

```suggestion
        def authenticate_pre_6x(password)
```

Still not perfect, but feels clearer

Reply:
renamed